include/serialport.h

```diff
@@ -51,28 +51,25 @@
 
 class MyFifo {
 public:
-	MyFifo(Bitu maxsize_) {
-		maxsize=size=maxsize_;
-		pos=used=0;
-		data=new Bit8u[size];
+	MyFifo(uint32_t maxsize_)
```

Comment:
Yeah, it should be safe to use 32-bit value here… But C++ containers tend to use size_t for this value. Also, if we're changing type here - then adopting STL naming convention would make sense (that would make constructor `MyFifo(size_t n)` or `MyFifo(size_t capacity)`.

Reply:
Ahh good point.   So if the argument is going to feed an STL containter size down the line, we will use `size_t` straight away. (And the 'capacity or n' naming). 

Will change this, and take a pass through in general to catch other cases like it.

Reply:
Adjusted the FIFO class to now use `size_t`s and an underlying `vector<uint8_t>` instead of a manually-managed array. Part of 48d12fae.